### ISSUE TITLE
LT01 false positives when using BigQuery concatenation operator

### ISSUE THREAD

**Issue by Blufe** [sqlfluff/sqlfluff#5818]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff does not correctly recognize BigQuery's string concatenation operator (||) as a binary operator.

### Expected Behaviour

The "sqlfluff lint" command succeeds without an error.

### Observed Behaviour

The command error with this output:

```
# echo 'SELECT "a" || "b" AS s FROM t;' | sqlfluff lint --config .sqlfluff
== [/opt/env/devintegrate/hoge.sql] FAIL                                                                                                                                                                    
L:   1 | P:  13 | LT01 | Expected single whitespace between pipe '|' and pipe
                       | '|'. [layout.spacing]
```

### How to reproduce

Run `echo 'SELECT "a" || "b" AS s FROM t;' | sqlfluff lint --config .sqlfluff`

In BigQuery, || is the string concatenation operator.
But this has been misidentified.
https://cloud.google.com/bigquery/docs/reference/standard-sql/operators

### Dialect

bigquery

### Version

3.0.5

### Configuration

[sqlfluff]
dialect = bigquery
rules = core

[sqlfluff:layout:type:binary_operator]
spacing_within = single
line_position = leading

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
I'm confused about this issue, and I think it's based on asking the linter to do something invalid:

By saying:

```
[sqlfluff:layout:type:binary_operator]
spacing_within = single
```

What you're saying is that _inside_ any binary operator, there should be a space: e.g. `< =`and not `<=`. Naturally this is invalid in _most_ (but not all sql dialects - e.g. oracle), but sqlfluff will allow that configuration if you want to set it.

The solution here is not to set that specific layout config for bigquery, as you're asking it to do something which bigquery doesn't actually support.
